lsp/systemd_lsp.lua

```diff
@@ -12,9 +12,58 @@
 --- ```
 ---
 --- A language server implementation for Systemd unit files made in Rust.
+---
+--- > [!NOTE]
+--- >
+--- > If you want this LSP to accurately activate for any Systemd files,
+--- > make sure to use the following autocmd:
+---
+--- ```lua
+--- vim.api.nvim_create_autocmd({ 'BufReadPost', 'BufNewFile' }, {
+---  group = vim.api.nvim_create_augroup('systemd-filetypes', { clear = true }),
+--   pattern = {
+--     '*.service',
+--     '*.mount',
+--     '*.device',
+--     '*.nspawn',
+--     '*.target',
+--     '*.timer',
+--     '*.path',
+--     '*.slice',
+--     '*.socket',
+--   },
+--   callback = function()
+--     local bufnr = vim.api.nvim_get_current_buf()
+--     vim.bo[bufnr].filetype = 'systemd'
+--   end,
+--   desc = 'Set filetype to systemd for systemd unit files',
+-- })
+--- ```
 
 ---@type vim.lsp.Config
 return {
   cmd = { 'systemd-lsp' },
   filetypes = { 'systemd' },
+  root_pattern = { -- Credit to @magnuslarsen
+    -- systemd unit files
+    '*.service',
```

Comment:
? Here is an example: https://github.com/neovim/nvim-lspconfig/blob/41ceb6bba3a40128b7841f5b7f5a9dae7201e823/lsp/ols.lua#L15

Reply:
Oh shoot, I realized I used it wrong. My bad, I thought it was a table field.